x-pack/solutions/security/plugins/security_solution/public/flyout/document_details/right/components/highlighted_fields.tsx

```diff
@@ -76,80 +81,126 @@ const columns: Array<EuiBasicTableColumn<HighlightedFieldsTableRow>> = [
+   */
+  isPreview: boolean;
+  /**
+   * Maintain backwards compatibility // TODO remove when possible
+   * Only needed if alerts page flyout (which uses CellActions), NOT in the AI for SOC alert summary flyout.
+   */
+  scopeId?: string;
+  /**
+   * If true, cell actions will be shown on hover.
+   * This is false by default (for the AI for SOC alert summary page) and will be true for the alerts page.
+   */
+  showCellActions?: boolean;
+}
+
 /**
  * Component that displays the highlighted fields in the right panel under the Investigation section.
+ * It is used in both in the alerts page and the AI for SOC alert summary page. The latter has no CellActions enabled.
  */
-export const HighlightedFields: FC = () => {
-  const { dataFormattedForFieldBrowser, scopeId, isPreview, investigationFields } =
-    useDocumentDetailsContext();
-
-  const [isEditLoading, setIsEditLoading] = useState(false);
-  const editHighlightedFieldsEnabled = useIsExperimentalFeatureEnabled(
-    'editHighlightedFieldsEnabled'
-  );
-
-  const highlightedFields = useHighlightedFields({
+export const HighlightedFields = memo(
+  ({
     dataFormattedForFieldBrowser,
     investigationFields,
-  });
-  const items = useMemo(
-    () => convertHighlightedFieldsToTableRow(highlightedFields, scopeId, isPreview),
-    [highlightedFields, scopeId, isPreview]
-  );
+    isPreview,
+    scopeId = '',
+    showCellActions = false,
+  }: HighlightedFieldsProps) => {
+    const [isEditLoading, setIsEditLoading] = useState(false);
+    const editHighlightedFieldsEnabled = useIsExperimentalFeatureEnabled(
+      'editHighlightedFieldsEnabled'
+    );
+
+    const highlightedFields = useHighlightedFields({
+      dataFormattedForFieldBrowser,
+      investigationFields,
+    });
+    const items = useMemo(
+      () =>
+        convertHighlightedFieldsToTableRow(highlightedFields, scopeId, isPreview, showCellActions),
+      [highlightedFields, scopeId, isPreview, showCellActions]
+    );
 
-  return (
-    <EuiFlexGroup direction="column" gutterSize="none">
-      <EuiFlexItem>
-        <EuiFlexGroup alignItems="center" css={{ minHeight: '40px' }}>
-          <EuiFlexItem data-test-subj={HIGHLIGHTED_FIELDS_TITLE_TEST_ID}>
-            <EuiTitle size="xxs">
-              <h5>
+    return (
+      <EuiFlexGroup direction="column" gutterSize="none">
+        <EuiFlexItem>
+          <EuiFlexGroup alignItems="center" css={{ minHeight: '40px' }}>
+            <EuiFlexItem data-test-subj={HIGHLIGHTED_FIELDS_TITLE_TEST_ID}>
+              <EuiTitle size="xxs">
+                <h5>
+                  <FormattedMessage
+                    id="xpack.securitySolution.flyout.right.investigation.highlightedFields.highlightedFieldsTitle"
+                    defaultMessage="Highlighted fields"
+                  />
+                </h5>
+              </EuiTitle>
+            </EuiFlexItem>
+            {editHighlightedFieldsEnabled && (
```

Comment:
do you want to allow user to customize highlighted fields?

Reply:
Nope, that is correct, I actually moved this logic to the parent to simplify, finishing unit tests then I'll push the change!